tribits/core/package_arch/TribitsInternalPackageWriteConfigFile.cmake

```diff
@@ -425,35 +351,201 @@ function(tribits_generate_package_config_file_for_install_tree  packageName)
+#     ``<packageConfigForBuildBaseDir>/`` (and any subdirs that does exist
+#     will be created).  The generated file ``<packageName>Config.cmake`` is
+#     for usage of the package in the build tree (not the install tree) and
+#     points to include directories and libraries in the build tree.
+#
+#   ``PACKAGE_CONFIG_FOR_INSTALL_BASE_DIR <packageConfigForInstallBaseDir>``
+#
+#     If specified, then the package's ``<packageName>Config_install.cmake``
+#     file and supporting files will be written under the directory
+#     ``<packageConfigForInstallBaseDir>/`` (and any subdirs that does exist
+#     will be created).  The file ``${PACKAGE_NAME}Config_install.cmake`` is
+#     meant to be installed renamed as ``<packageName>Config.cmake`` in the
+#     install tree and it points to installed include directories and
+#     libraries.
+#
+# NOTE: This function does *not* contain any ``install()`` command itself
+# because CMake will not allow those to even be present in scripting mode that
+# is used for unit testing this function.  Instead, the commands to install
+# the files are added by the function
+# ``tribits_write_package_client_export_files_install_targets()``.
+#
+function(tribits_write_flexible_package_client_export_files)
 
   if (TRIBITS_WRITE_FLEXIBLE_PACKAGE_CLIENT_EXPORT_FILES_DEBUG_DUMP)
-    message("tribits_append_dependent_package_config_file_includes_and_enables(${ARGV})")
+    message("\ntribits_write_flexible_package_client_export_files(${ARGN})")
   endif()
 
-  # Parse input
+  #
+  # A) Process the command-line arguments
+  #
 
   cmake_parse_arguments(
-     PARSE ""  # prefix, options
-     #one_value_keywords
-     "EXPORT_FILE_VAR_PREFIX;EXT_PKG_CONFIG_FILE_BASE_DIR;PKG_CONFIG_FILE_BASE_DIR;CONFIG_FILE_STR_INOUT"
-     "" #multi_value_keywords
+     #prefix
+     PARSE
+     #options
+     "WRITE_INSTALL_CMAKE_CONFIG_FILE"
+     #one_value_keywords
+     "PACKAGE_NAME;EXPORT_FILE_VAR_PREFIX;PACKAGE_CONFIG_FOR_BUILD_BASE_DIR;PACKAGE_CONFIG_FOR_INSTALL_BASE_DIR"
+     #multi_value_keywords
+     ""
+     ${ARGN}
+     )
+
+  tribits_check_for_unparsed_arguments()
+
+  if (NOT ${PROJECT_NAME}_GENERATE_EXPORT_FILE_DEPENDENCIES)
+    message(SEND_ERROR "Error: Can't generate export dependency files because"
+      " ${PROJECT_NAME}_GENERATE_EXPORT_FILE_DEPENDENCIES is not ON!")
+    return()
+  endif()
+
+  set(PACKAGE_NAME ${PARSE_PACKAGE_NAME})
+  if (TRIBITS_WRITE_FLEXIBLE_PACKAGE_CLIENT_EXPORT_FILES_DEBUG_DUMP)
+    print_var(PACKAGE_NAME)
+  endif()
+
+  set(EXPORT_FILE_VAR_PREFIX ${PACKAGE_NAME})
+  if (PARSE_EXPORT_FILE_VAR_PREFIX)
+    set(EXPORT_FILE_VAR_PREFIX ${PARSE_EXPORT_FILE_VAR_PREFIX})
+  endif()
+  if (TRIBITS_WRITE_FLEXIBLE_PACKAGE_CLIENT_EXPORT_FILES_DEBUG_DUMP)
+    print_var(EXPORT_FILE_VAR_PREFIX)
+  endif()
+
+  # Generate a note discouraging editing of the <package>Config.cmake file
+  set(DISCOURAGE_EDITING "Do not edit: This file was generated automatically by CMake.")
```

Comment:
It is here:

https://github.com/TriBITSPub/TriBITS/blob/f58975fd8a9b6df6eaec7ba83e8ad6864a43d73d/tribits/core/installation/TribitsPackageConfigTemplate.cmake.in#L44

I did not write the initial implementation of this code.  My guess is that the original author did not want to put that comment in the file `TribitsPackageConfigTemplate.cmake.in` because CMake did **not** generate that file.